tests/blackbox_test.go

```diff
@@ -0,0 +1,1215 @@
+		"-i", "4096", "-I", "128", partition.Device).CombinedOutput()
+	if err != nil {
+		t.Fatal("mke2fs", err, string(out))
+	}
+
+	opts := []string{"-e", "remount-ro"}
+	if partition.Label != "" {
+		opts = append(opts, "-L", partition.Label)
+	}
+
+	if partition.TypeCode == "coreos-usr" {
+		opts = append(
+			opts, "-U", "clear", "-T", "20091119110000", "-c", "0", "-i", "0",
+			"-m", "0", "-r", "0")
+	}
+	opts = append(opts, partition.Device)
+	tuneOut, err := exec.Command("/sbin/tune2fs", opts...).CombinedOutput()
+	if err != nil {
+		t.Fatal("tune2fs", err, string(tuneOut))
+	}
+}
+
+func formatBTRFS(t *testing.T, partition *Partition) {
+	opts := []string{}
+	if partition.Label != "" {
+		opts = append(opts, "--label", partition.Label)
+	}
+	opts = append(opts, partition.Device)
+	out, err := exec.Command("/sbin/mkfs.btrfs", opts...).CombinedOutput()
+	if err != nil {
+		t.Fatal("mkfs.btrfs", err, string(out))
+	}
+}
+
+func align(count int, alignment int) int {
+	offset := count % alignment
+	if offset != 0 {
+		count += alignment - offset
+	}
+	return count
+}
+
+func setOffsets(partitions []*Partition) {
+	offset := 34
+	for _, p := range partitions {
+		if p.Length == 0 || p.TypeCode == "blank" {
+			continue
+		}
+		offset = align(offset, 4096)
+		p.Offset = offset
+		offset += p.Length
+	}
+}
+
+func createPartitionTable(
+	t *testing.T, fileName string, partitions []*Partition) {
+	opts := []string{fileName}
+	hybrids := []int{}
+	for _, p := range partitions {
+		if p.TypeCode == "blank" || p.Length == 0 {
+			continue
+		}
+		opts = append(opts, fmt.Sprintf(
+			"--new=%d:%d:+%d", p.Number, p.Offset, p.Length))
+		opts = append(opts, fmt.Sprintf(
+			"--change-name=%d:%s", p.Number, p.Label))
+		if p.TypeGUID != "" {
+			opts = append(opts, fmt.Sprintf(
+				"--typecode=%d:%s", p.Number, p.TypeGUID))
+		}
+		if p.GUID != "" {
+			opts = append(opts, fmt.Sprintf(
+				"--partition-guid=%d:%s", p.Number, p.GUID))
+		}
+		if p.Hybrid {
+			hybrids = append(hybrids, p.Number)
+		}
+	}
+	if len(hybrids) > 0 {
+		if len(hybrids) > 3 {
+			t.Fatal("Can't have more than three hybrids")
+		} else {
+			opts = append(opts, fmt.Sprintf("-h=%s", intJoin(hybrids, ":")))
+		}
+	}
+	sgdiskOut, err := exec.Command(
+		"/sbin/sgdisk", opts...).CombinedOutput()
+	if err != nil {
+		t.Fatal("sgdisk", err, string(sgdiskOut))
+	}
+}
+
+func kpartxAdd(t *testing.T, fileName string) string {
+	kpartxOut, err := exec.Command(
+		"/sbin/kpartx", "-av", fileName).CombinedOutput()
+	if err != nil {
+		t.Fatal("kpartx", err, string(kpartxOut))
+	}
+	kpartxOut, err = exec.Command(
+		"/sbin/kpartx", "-l", fileName).CombinedOutput()
+	return strings.Trim(strings.Split(string(kpartxOut), " ")[4], "/dev/")
+}
+
+func mountRootPartition(t *testing.T, partitions []*Partition) {
+	for _, partition := range partitions {
+		if partition.Label != "ROOT" {
+			continue
+		}
+		mountOut, err := exec.Command(
+			"/bin/mount", partition.Device,
```

Comment:
Tiny nit, feel free to ignore, for things like this I generally prefer to use [the `syscall` package](https://golang.org/pkg/syscall/#Mount) instead of exec'ing something else. Not really an issue here because this entire thing is already dependent on a bunch of other binaries.